File: apm/src/main/java/io/github/yemyatthu1990/apm/Mapper.java
package io.github.yemyatthu1990.apm;

import androidx.annotation.Nullable;

import java.util.Collection;
import java.util.Set;
import java.util.concurrent.ConcurrentMap;

interface Mapper {
    int size();
    boolean isEmpty();
    boolean hasKey(String key);
    @Nullable
    String getValue(String key);
    ConcurrentMap<String, String> map();
    Set<String> getKeySet();
    Collection<String> valueSet();
    void put( String key, String value);
    void putAll( ConcurrentMap<String, String> map);
    void clear();
}

File: apm/src/main/java/io/github/yemyatthu1990/apm/collectors/Mapper.kt
package io.github.yemyatthu1990.apm.collectors

import java.util.concurrent.ConcurrentMap

interface Mapper {
    fun size(): Int
    val isEmpty: Boolean
    fun hasKey(key: String): Boolean
    fun getValue(key: String): String?
    fun map(): ConcurrentMap<String, String>?
    val keySet: Set<String>?
    fun valueSet(): Collection<String>?
    fun put(key: String, value: String)
    fun putAll(map: ConcurrentMap<String, String>?)
    fun clear()
}